docs/developer-docs/developer-tools/off-chain/agents/rust-agent.mdx

```diff
@@ -24,6 +24,104 @@ To add the `ic-agent` crate as a dependency in your project, use the command:
 cargo add ic-agent
 ```
 
+## Initializing the agent
+
+Before using the agent in your project, it must be initialized using the `Agent::builder()` function. This can be done within your Rust code, or in a separate `agent.rs` file that can be imported into other Rust files. Here is an example of how to initialize the Rust agent:
+
+```rust
+use anyhow::Result;
+use ic_agent::{agent::http_transport::reqwest_transport::ReqwestHttpReplicaV2Transport, Agent};
+
+pub async fn create_agent(url: &str) -> Result<Agent> {
+    let transport = ReqwestHttpReplicaV2Transport::create(url)?;
+
+    let agent = Agent::builder().with_transport(transport).build()?;
+    agent.fetch_root_key().await?;
+
+    Ok(agent)
+}
+```
+
+
+## Authentication
+
+The Rust agent's `Identity` object provides signatures that can be used for HTTP requests or identity delegations. It represents the principal ID of the sender.
+
+Identities can be created when the agent is initialized using the `.with_identity(create_identity())` method. Each agent has a single `Identity` object that can contain multiple identity values.
+
+```rust
+async fn create_a_canister() -> Result<Principal, Box<dyn std::error::Error>> {
+  let agent = Agent::builder()
+    .with_url(URL)
+    .with_identity(create_identity())
+    .build()?;
+```
+
+
+## Making calls
+
+The Rust agent can be used to make calls to other canisters. To make a call to a canister, use the `agent.update` method for an update call, or `agent.query` for a query call. Then, pass in the canister's ID and the method of the canister you'd like to call.
+
+For example, to call the management canister's `provisional_create_canister_with_cycles` method, use the following code:
+
+```rust
+  let effective_canister_id = Principal::from_text("rwlgt-iiaaa-aaaaa-aaaaa-cai").unwrap();
+  let response = agent.update(&management_canister_id, "provisional_create_canister_with_cycles")
```

Comment:
This block is probably misleading on its own as it indicates the `with_effective_canister_id` method, which is only meaningful when calling management canister functions. This should probably have a more conventional canister call above it:
```rs
#[derive(CandidType)]
struct AccountBalanceArgs {
    account: Vec<u8>,
}
#[derive(CandidType, Deserialize)]
struct Tokens {
    e8s: u64,
}
let icp_ledger = Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
let response = agent.update(&icp_ledger, "account_balance")
    .with_arg(Encode!(&AccountBalanceArgs { account: some_account_id })?)
    .call_and_wait()
    .await?;
let tokens = Decode!(&response, Tokens)?;
```

Reply:
Replaced example